File: frontend/src/types/learning.ts
export interface LearningPath {
  id: number;
  title: string;
  description: string;
  icon: string;
  level: 'Beginner' | 'Intermediate' | 'Advanced' | 'Expert';
  progress: number;
  lessons: number;
  completed: number;
  courses: Course[];
}

export interface Course {
  id: number;
  pathId: number;
  title: string;
  description: string;
  duration: string;
  completed: boolean;
  order: number;
  type: 'video' | 'article' | 'project' | 'quiz';
}

export interface Project {
  id: number;
  title: string;
  description: string;
  githubUrl?: string;
  difficulty: 'Beginner' | 'Intermediate' | 'Advanced';
  technologies: string[];
  completed: boolean;
}

export interface ProgressStats {
  enrolledTracks: number;
  lessonsCompleted: number;
  practiceHours: number;
  achievements: number;
}

File: backend/app/schemas/learning.py
from pydantic import BaseModel
from typing import Optional, List

class Course(BaseModel):
    id: int
    path_id: int
    title: str
    description: Optional[str] = None
    duration: Optional[str] = None
    completed: bool = False
    order: int = 0
    type: str = "article"  # video, article, project, quiz

    class Config:
        from_attributes = True

class LearningPath(BaseModel):
    id: int
    title: str
    description: Optional[str] = None
    icon: Optional[str] = None
    level: str = "Beginner"
    progress: float = 0.0
    lessons: int = 0
    completed: int = 0
    courses: List[Course] = []

    class Config:
        from_attributes = True

class ProgressResponse(BaseModel):
    enrolled_tracks: int = 0
    lessons_completed: int = 0
    practice_hours: float = 0.0
    achievements: int = 0
